### DISCUSSION TITLE
12/20/22 Standup Discussion

### DISCUSSION THREAD

**Discussion by Caleb-Cohen** [Together-100Devs/Together#193] [General]
# 12/20/22 Standup (Improvised. Adapt. Overcome Version):

Thanks to everyone that’s put some time into Together since last Thursday! I know things are a bit hectic with the holidays, but it’s so awesome how Together is coming along with all of us working asynchronously. 

To organize, please try to keep sub-discussion threads directly related to that specific issue.

## Updates since last Thursday

- Include discordName with an Event - COMPLETED by @romanstetsyk 
- Add more instructions for beginners into the wiki/contributing documents - COMPLETED by @Caleb-Cohen & @intelagense 
- Event view modal is not clickable when the user is not logged in - COMPLETED by @MatthewBozin 

## Open Issues

@TreLynnnn, I left some comments about resizing the smaller logout on your PR #187. Do you have an update on that?

@WonjaeSung @isaaclee12  - How’s the soft security coming? I know we had some comments in the group chat earlier on the approach. Is there any more discussion needed on that? Is there anything you need from me to continue? 

@MatthewBozin, @JoeDravarol & @DevRhyse, Where do we stand on the context redesign? I see that issue #192 was posted yesterday. Is there anything you can add here?

@luciousmc & @HondaChan14 Do you have any more comments on @Natashka333 design? I left some comments on my favorites, but I'd love to see what y'all think.

This may be better for our next voice standup, but what are our thoughts on #181? I like the concept of the Number 3 that @miguelbj mentioned. @intelagense, where do you stand? I want to get a clear path to move forward on this work without rushing it and not sitting on it forever.   

## Next Standup? 

Plan for the next standup to be of a similar format, considering the 100Devs class and I will be traveling. Thanks for everything that y'all do! 

## Feedback?

We love it; what you got?

## Riddle!

Who's got last week's answer?

Here is todays! 👽☎️🏠 (No spoiling it Joseph, Joseph, or Joseph)

**Comment by isaaclee12** [comment]
Hi @Caleb-Cohen! Soft security is going along steady. We met Monday as well as today and got the basic functionality done.

I'm tight for time so I can't do a thorough explanation, but the long and short of it is that the 90 day limit, the 2022-2023 boundaries, and prevention of selecting a date before today is functional, though there are some logic errors (e.g. you can set the end date to be before the start date, though this doesn't seem to break anything so far).

You can check out the code here: https://github.com/isaaclee12/Together/tree/feature/softSecurity

Or you can pull it by running ```git remote set-url origin https://github.com/isaaclee12/Together.git``` in your cmd
then running ```git pull origin feature/softSecurity```.


API stuff as well as further issues with the forms behavior are yet to be implemented, will likely be speaking with Matt and others before fixing these issues.

**Reply by Caleb-Cohen** [reply]
Thanks for the update!

In terms of the rejection, does Together notify the user, or does it just reject input?

Glad to hear the front-end block, like the 90-day limit and the year boundaries, is coming along.

API stuff like the 5 events per user will be more difficult. I'll add it to the standup discussion.

**Reply by isaaclee12** [reply]
At the moment, Together just outright doesn't allow the actual end or start date to be set beyond the 90-day, year, or today boundaries 
(i.e. the dates are disabled on the calendar selector for each field).

We've realized that, while this is functional, it may actually be bad practice as the user might get frustrated with only ever being able to set the start or end date within 90 days of each other.

An example of this:
- The user sets the start date to 01/01/2023, then sends end date to 04/01/2023 (which is 90 days after 01/01/2023).
- The user decides the want to change the recurrence range to 02/01/2023 - 05/02/2023 (also a 90 day gap).
- Because 05/02/2023 is more than 90 days from our current start date, 01/01/2023, the user must then change the START DATE to 02/01/2023 first before being able to move the end date forward.

Let us know your thoughts, we can change the functionality before PR'ing.

**Reply by isaaclee12** [reply]
Furthermore, I've been taking a look at the express API now and I'm definitely seeing that there might need to be a new table added to the schema (and/or some data normalization). Would love to discuss during standup tomorrow.

**Comment by JoeDravarol** [comment]
Hey @Caleb-Cohen, in regards to refactoring context, we still need time to discuss as everyone has been busy over the festive season. Currently, I have a sandbox to demonstrate the general idea of how we should seperate context so that's it's more maintainable and easier to add, modify and extend features. [Definitely not a terrible sandbox](https://codesandbox.io/s/stupefied-babycat-bsrchd)

**Reply by Caleb-Cohen** [reply]
> Hey @Caleb-Cohen, in regards to refactoring context, we still need time to discuss as everyone has been busy over the festive season. Currently, I have a sandbox to demonstrate the general idea of how we should seperate context so that's it's more maintainable and easier to add, modify and extend features. [Definitely not a terrible sandbox](https://codesandbox.io/s/stupefied-babycat-bsrchd)

understood. I plan on joining your discussion with @MatthewBozin regarding this today.

**Comment by intelagense** [comment]
for #181 and any other backend issues, we should agree on the features that will be added to the next major release and extend both the event model and the user model to accommodate.

**Reply by Caleb-Cohen** [reply]
> for #181 and any other backend issues, we should agree on the features that will be added to the next major release and extend both the event model and the user model to accommodate.

I agree.
